contributed_definitions/nyaml/NXfit.yaml

```diff
@@ -0,0 +1,263 @@
+category: base
+doc: |
+  Description of a fit procedure.
+symbols:
+  doc: |
+    The symbols used in the schema to specify e.g. dimensions of arrays.
+  n_points: |
+    Number of discretization points inside the fitting region.
+  n_independent: |
+    Number of independent axes for this fit procedure.
+type: group
+NXfit(NXobject):
+  label:
+    doc: |
+      Human-readable label for this fit procedure.
+  data(NXdata):
+    doc: |
+      Input data and results of the fit.
+    input_dependent(NX_NUMBER):
+      unit: NX_ANY
+      doc: |
+        Dependent axis for this fit procedure.
+      dimensions:
+        dim: [[1, n_points]]
+    input_independent(NX_NUMBER):
+      unit: NX_ANY
+      doc: |
+        Independent axis for this fit procedure.
+      dimensions:
+        dim: [[1, n_independent], [2, n_points]]
+    envelope(NX_NUMBER):
+      unit: NX_ANY
+      doc: |
+        Resulting envelope of all NXpeak instances.
+        Should have the same dimension as the `position` axes of the individual peaks.
+      dimensions:
+        dim: [[1, n_independent], [2, n_points]]
+    residual(NX_NUMBER):
+      unit: NX_ANY
+      doc: |
+        Difference between the envelope and the data to be fitted.
+      dimensions:
+        dim: [[1, n_independent], [2, n_points]]
+  peakPEAK(NXpeak):
+    doc: |
+      One spectral peak of the peak model.
+      It is envisioned that peaks are labeled as peak_0, peak_1, and so on.
+    total_area(NX_NUMBER):
+      unit: NX_ANY
+      doc: |
+        Total area under the curve (can also be used for the total area minus any
+        background values).
+    relative_sensitivity_factor(NX_NUMBER):
+      unit: NX_UNITLESS
+      doc: |
+        Relative sensitivity for this peak, to be used for quantification in
+        an NXprocess.
+        
+        As an example, in X-ray spectroscopy could depend on the energy scale
+        (see position), the ionization cross section, and the element probed.
+    relative_area(NX_NUMBER):
+      unit: NX_ANY
+      doc: |
+        Relative area of this peak compared to other peaks.
+        
+        The relative area can simply be derived by dividing the total_area by the
+        total area of all peaks or by a more complicated method (e.g., by
+        additionally dividing by the relative sensitivity factors). Details shall
+        be given in fitting_method.
+  backgroundBACKGROUND(NXfit_background):
+    doc: |
+      One fitted background (functional, position, and intensities) of the peak fit.
+      It is envisioned that peaks are labeled as background_0, background_1, and
+      so on.
```

Comment:
Shouldn't peak and background be part of an NXfit_function?

Reply:
The idea was that you define your peaks/backgrounds individually and then the fit_function is just the formula for how the fit across all these peaks is done (e.g., least_squares). Maybe actually it is not great to use NXfit_function in two different contexts though,